clientv3/balancer.go

```diff
@@ -222,11 +214,14 @@ func (b *simpleBalancer) Close() error {
 	}
 	b.closed = true
 	close(b.notifyCh)
-	// terminate all waiting Get()s
-	b.pinAddr = ""
-	if len(b.upEps) == 0 {
+	select {
+	case <-b.upc: // do nothing if upc has been closed
+	default:
+		// close upc if it has not been closed
+		// terminate all waiting Get()s
```

Comment:
if is possible that b.upc is already closed before `Close()` is called. Calling `close(b.upc)` twice causes panic. I implemented your suggestion before, and I observed the panic.